### PR TITLE
Task 4

### PR DESCRIPTION
**Pull request by helly-fox**
## Task 4

- SQL script is created and placed in repo (databased filled already with data) - fill-db-with-data.sql
- all lamdba are switched to usage of DRS instead of mock data (mock json was removed)
- POST endpoint is implemented
- Frond end is using DB (could be checked by posting new item in DB and verify if it appears on front end)
- POST/products lambda functions returns error 400 status code if product data is invalid (JOI package was used) (Small ask to add smth relevant 🎄 )
- All lambdas return error 500 status code on any error (DB connection, any unhandled error in code)
- All lambdas do console.log for each incoming requests and their arguments
- Transaction based creation of product

Link for API https://nkmnmeycd8.execute-api.eu-west-1.amazonaws.com/dev/products
Link for cloudfront https://d22ps2engidd1v.cloudfront.net/
FE repo pr https://github.com/helly-fox/nodejs-aws-fe/pull/4/files

### CODE REVIEW

**Comment by helly-fox** [comment]
⏰  Releasing new version (could be down for now)

**Comment by helly-fox** [comment]
> ⏰ Releasing new version (could be down for now)

Live again

**Comment by lithiumfallen** [comment]
Missed `s` in request from fe side
https://d22ps2engidd1v.cloudfront.net/admin/products
https://d22ps2engidd1v.cloudfront.net/admin/product-form/
BE side works

File: product-service/fill-db-with-data.sql [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+insert into stocks (product_id, count) values
+('7567ec4b-b10c-48c5-9345-fc73c48a80aa', 4),
+('7567ec4b-b10c-48c5-9345-fc73c48a80a0', 6),
+('7567ec4b-b10c-48c5-9345-fc73c48a80a2', 7),
+('7567ec4b-b10c-48c5-9345-fc73c48a80a1', 12),
+('7567ec4b-b10c-48c5-9345-fc73c48a80a3', 7),
+('7567ec4b-b10c-48c5-9345-fc73c48a80a4', 8),
+('7567ec4b-b10c-48c5-9345-fc73c48a80a5', 2),
+('7567ec4b-b10c-48c5-9345-fc73c48a80a6', 3);
```
Comment (lithiumfallen) [inline product-service/fill-db-with-data.sql:35]: JFYI. Using `uuid-ossp` you can skip manual id in your initial data

**Comment by helly-fox** [comment]
> Missed `s` in request from fe side
> https://d22ps2engidd1v.cloudfront.net/admin/products
> https://d22ps2engidd1v.cloudfront.net/admin/product-form/
> BE side works

Sorry just understood that we can bind it (updated on cloud stack will update FE soon)

File: product-service/fill-db-with-data.sql (hunk 1 above)
Comment (skyRoma) [reply product-service/fill-db-with-data.sql:35]: @lithiumfallen but for initial data we need the same ID's in both tables. Will this extension works correctly with it? 
image

**Review by skyRoma — COMMENTED** [review]
image

https://github.com/jquense/yup

**Review by skyRoma — APPROVED** [review]

**Pull request merged**